c7n/resources/sagemaker.py

```diff
@@ -107,6 +107,53 @@ def _augment(j):
             return list(filter(None, w.map(_augment, jobs)))
 
 
+@resources.register('sagemaker-transform-job')
+class SagemakerTransformJob(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_transform_jobs', 'TransformJobSummaries', None)
+        detail_spec = (
+            'describe_transform_job', 'TransformJobName', 'TransformJobName', None)
+        id = 'TransformJobArn'
+        name = 'TransformJobName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = None
```

Comment:
`NameContains`? for doing lambda policies its important the resource supports some notion of filtering either via filter_name, or the resource manager implements get_resources method.

Reply:
added filter_name